AWGL/somatic_db
# LIMS XML and bug fixes

## Change description
- Download an XML for LIMS from clicking dropdown that's now part of the download PDF button on the Report tab #60 
- Add page numbers to PDF, uses different package so env has changed, also adjusted formatting #13
- Fix issue where the wrong tab is defaulted to #47 
- Fix bug where you couldnt reopen a sample fail or enter a fail 2nd check if you're assigned to it, also tidy up of that page because there's lots of different options and there were loads of nested loops in template
- Add 'No gaps' or 'No calls' to results page when there are no reportable calls/gaps (both GUI and PDF, not needed for XML), for both DNA and RNA
- Fixed visualisation of COSMIC percentages and added columns for COSMIC counts in Depth & Report tabs & PDF (for solid tumour referrals only)
- Added 'options' page to link out to various other things such as poly lists (and probably other things in future
- Split the poly list into separate build 37 and 38 pages (you'll need to add the two poly lists on your test dev, called `build_37_polys` and `build_38_polys`) 

## Justification for minor update
No effect on processing of data, only how it's visualised

## Testing description
- Running existing unit tests
- LIMS XML not needed yet, will be tested as part of LIMS validation, may be tweaked as we go along, just review the code
- COSMIC counts/percents will be tested more in future PR - need test data from Sophie, just review the code
- Manual testing of database functionality
  - XML downloads and data matches report tab
  - PDF numbers pages correctly, data matches report tab and formatting is okay
  - No calls/ No gaps shown when no calls in report tab and PDF, instead of empty table
  - tried all possible combinations for the assigned to buttons on the samples page, shown in the comments of `view_samples.html`
  - Tested that text after hash is successfully removed when going onto the `view_samples.html` page (analyse one run normally and when you're redirected back there should be no hash)
  - Test adding polys to the build 37 and build 38 lists and make sure that they end up in the correct ones

## Test data location
N/A

## Verification results
- Unit tests all completed successfully
- Database testing listed above all performed as expected
- Deployment plan:
  - Simple model change - defaults to null
  - Notify CS about PDF changes in case there's any weird formatting that needs fixing on the fly
  - Cant specify output location for PDF/XML due to security issues, but can show CS how to set up browser to save as rather than just download straight into download folder  
  - Conda environment will need updating, test with different env name first as Viglen conda is temperamental
